ext/hash/hash.c

```diff
@@ -1529,8 +1552,14 @@ PHP_METHOD(HashContext, __unserialize)
 
 	hash->ops = ops;
 	hash->context = php_hash_alloc_context(ops);
-	ops->hash_init(hash->context);
 	hash->options = options;
+	if (args_zv && IS_ARRAY == Z_TYPE_P(args_zv)) {
+		ops->hash_init(hash->context, Z_ARRVAL_P(args_zv));
+		hash->args = zend_array_dup(Z_ARRVAL_P(args_zv));
```

Comment:
Why does this duplicate the array?

Reply:
`args_zv` is from the HT passed to ZPP, so the original will be destroyed. But `hash->args` needs to survive. I'll double check on this.

Thanks.

Reply:
Right. However, I don't think you need to actually copy the array, just incrementing the refcount should be enough. It doesn't get modified, right?

Reply:
Yeah, it should not be modified. Refcount++ is a great idea, will definitely follow up there. Perhaps also the struct should have this member as `const`, but i'll at least comment there that it should be expected to be RO.

Thanks!

Reply:
Fixed, working with GC_ADDREF now. 

Thanks.